cluster-autoscaler/cloudprovider/clusterapi/clusterapi_unstructured.go

```diff
@@ -118,17 +121,18 @@ func (r unstructuredScalableResource) SetSize(nreplicas int) error {
 		return err
 	}
 
-	s, err := r.controller.managementScaleClient.Scales(r.Namespace()).Get(context.TODO(), gvr.GroupResource(), r.Name(), metav1.GetOptions{})
-	if err != nil {
-		return err
+	spec := autoscalingv1.Scale{
+		Spec: autoscalingv1.ScaleSpec{
+			Replicas: int32(nreplicas),
+		},
 	}
 
-	if s == nil {
-		return fmt.Errorf("unknown %s %s/%s", r.Kind(), r.Namespace(), r.Name())
+	patch, err := json.Marshal(spec)
+	if err != nil {
+		return fmt.Errorf("could not marshal json patch for scaling: %w", err)
 	}
 
-	s.Spec.Replicas = int32(nreplicas)
-	_, updateErr := r.controller.managementScaleClient.Scales(r.Namespace()).Update(context.TODO(), gvr.GroupResource(), s, metav1.UpdateOptions{})
+	_, updateErr := r.controller.managementScaleClient.Scales(r.Namespace()).Patch(context.TODO(), gvr, r.Name(), types.MergePatchType, patch, metav1.PatchOptions{})
 
 	if updateErr == nil {
 		updateErr = unstructured.SetNestedField(r.unstructured.UnstructuredContent(), int64(nreplicas), "spec", "replicas")
```

Comment:
should we replace `spec` with `patch` in the error object here because that's what we're actually passing in now?